### ISSUE TITLE
Does this work on Laravel Vapor?

### ISSUE THREAD

**Issue by tabacitu** [Laravel-Backpack/basset#36]
👀

**Comment by malek77z** [comment]
Also, have the same question. I've completed an upgrade from backpack 5 to 6, which works perfectly locally, however, can't get this to work in Vapor.

**Comment by pxpm** [comment]
Hey @malek77z thanks for the info. 

Mind sharing the errors/troubles you are facing ? 

Cheers

**Comment by malek77z** [comment]
@pxpm 

yes for sure, so I have almost gotten there tbh but seems to be a bug or issue on the last step.
1. I changed to using s3 in the basset.php config... actually I changed my config to the following
```
return [
    // development mode, assets will not be internalized
    'dev_mode' => env('BASSET_DEV_MODE', env('APP_ENV') === 'local'),

    // disk and path where to store bassets
    'disk' => env('BASSET_DISK')??'public',
    'path' => 'basset',

    // use cache map file (.basset)
    'cache_map' => env('BASSET_CACHE_MAP')??true,

    // view paths that may use @basset
    // used to internalize assets in advance with artisan basset:internalize
    'view_paths' => [
        resource_path('views'),
    ],

    // content security policy nonce
    'nonce' => null,
];
```

This way I have more control of the variables on the vapor setup.
To note, the cache_map instantly breaks with this error: **"Could not find Backpack view [backpack.ui::layouts.] in theme namespace, fallback namespace nor UI namespace." // vendor/backpack/crud/src/helpers.php:259**
However, without it, it starts to write to S3.

To be able to see the files on S3 I had to update my s3 filesystems.php s3 config to have 'visibility' => 'public'
(it might be good to update the ->put in BassetManager to all be 'public' as well.. but could make another config in my filesystems.php config so not a dealbreaker)

It gets very close, however, the last issue seems to be with the way it get the path and cleans it.
i.e. in my code, it has a double '//' like so

```
<link href="https://xyz.amazonaws.com/basset//vendor/backpack/theme-coreuiv2/resources/assets/css/coreuiv2.css?37e27744030e" rel="stylesheet" type="text/css" />
```

Which is to do with the face that 
```
base_path('vendor/backpack/theme-coreuiv2/resources/assets/css/coreuiv2.css')  = "/var/task/vendor/backpack/theme-coreuiv2/resources/assets/css/coreuiv2.css"
```
and
```
base_path() = "/var/task"
```

So when the url get created from class BassetManager

```
$path = $this->getPath(is_string($output) ? $output : $asset);
```

It returns
```
"basset//vendor/backpack/theme-coreuiv2/resources/assets/css/coreuiv2.css"
```

If getPath was updated to something like the following

```
public function getPath(string $asset): string
    {
        return Str::of($this->basePath)
            ->append(str_replace([base_path().'/',base_path(), 'http://', 'https://', '://', '<', '>', ':', '"', '|', "\0", '*', '`', ';', "'", '+'], '', $asset))
            ->before('?')
            ->replace('/\\', '/');
    }
```

It would fix it... working out now how to pollyfill around this. If I'm doing something wrong would love to know how to address it.

Also, just to note to help anyone else, if you put s3 in the basset config by default on deploy, at least with my github action, it errors as looks for the s3 config that has not been created yet by Vapor. So therefor I added it to a config in Vapor I could switch out.
.

**Comment by malek77z** [comment]
@pxpm 
Sorry im still really a novice at github etc, but forked, deployed and confirmed it works:
https://github.com/malek77z/basset
If any of this is useful im very happy to have helped.

**Comment by tabacitu** [comment]
I am 100% sure it will be useful @malek77z , thank you very much. I've asked @promatik to take this over since he's the author of Basset.

I've created a PR from your repo here - https://github.com/Laravel-Backpack/basset/pull/57 - it's very clean PR btw, love it 👏👏👏

Antonio will take a look ASAP to push it forward or merge.

Thanks for the help.
Cheers!

**Comment by tabacitu** [comment]
Fixed in https://github.com/Laravel-Backpack/basset/pull/58 🤞 Thank again @malek77z - as I said in that PR too 😀
